clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/AbstractStorageEngine.java

```diff
@@ -340,6 +342,7 @@ public synchronized Map<String, String> sync(int partitionId) {
 
   @Override
   public synchronized void close() throws VeniceException {
+    this.closed = true;
```

Comment:
This may not be enough... for this strategy to work, the flag probably needs to be made true in `drop()` as well. I wonder though... is it possible for a storage engine to be dropped, and then to re-add more partitions into it again? If that is the case, then maybe there needs to be a code path that sets `closed` back to false also 🤔 ...

Reply:
Maybe we should check whether `partitionList` is empty or not instead of introducing a new flag?

Reply:
Good idea... I did that in the latest commit. LMK how it looks.